python/neuroglancer/ingest/tasks.py

```diff
@@ -347,6 +508,135 @@ def _encode(self, chunk, encoding):
+        return "HyperSquareTask(chunk_path='{}, chunk_encoding='{}', version='{}', info_path='{}')".format(
+            self.chunk_path, self.chunk_encoding, self.version, self.info_path)
+
+    def execute(self):
+        self._parse_info_path()
+        self._parse_chunk_path()
+        self._storage = Storage(self._dataset_name, self._layer_name, compress=False)
+        self._download_metadata()
+        self._download_data()
+        self._upload_chunk()
+
+    def _parse_chunk_path(self):
+        if 'segmentation' in self._layer_name: 
+            match = re.match(r'^gs://(.*)/(.*)/segmentation.lzma', self.chunk_path)
+            self._bucket_name, self._chunk_folder = match.groups()
+        elif 'image' in self._layer_name:
+            match = re.match(r'^gs://(.*)/(.*)/jpg/0\.jpg', self.chunk_path)
+            self._bucket_name, self._chunk_folder = match.groups()
+        else:
+            return NotImplementedError("Don't know how process this layer")
+
+
+    def _parse_info_path(self):
+        match = re.match(r'^.*/([^//]+)/([^//]+)/info$', self.info_path)
+        self._dataset_name, self._layer_name = match.groups()
+
+    def _download_metadata(self):
+        self._bucket = self._storage._client.get_bucket(self._bucket_name)
+        metadata = self._bucket.get_blob(
+            '{}/metadata-fixed.json'.format(self._chunk_folder)) \
+        .download_as_string()
+        self._metadata = json.loads(metadata)
+        
+    def _download_data(self):
+        if 'segmentation' in self._layer_name: 
+            self._datablob = self._bukcet.get_blob(
+                '{}/segmentation.lzma'.format(self._chunk_folder))
+            string_data = self._datablob.download_as_string()
+            self._data = self._decode_lzma(string_data)
+        elif 'image' in self._layer_name:
+            self._data = np.zeros(shape=(256,256,256), dtype=np.uint8) #x,y,z,channels
+            for blob in self._bucket.list_blobs(prefix='{}/jpg'.format(self._chunk_folder)):
+                z = int(re.findall(r'(\d+)\.jpg', blob.name)[0])
+                self._data[:,:,z] = chunks.decode_jpeg(blob.download_as_string()).transpose()
+            self._data.transpose((2,1,0))
+        else:
+            return NotImplementedError("Don't know how to get the images for this layer")
+
+    def _decode_lzma(self, string_data):
+        arr = lzma.decompress(string_data)
+        if self._metadata['segment_id_type'] == 'UInt8':
+            arr = np.fromstring(arr, dtype=np.uint8)
+        elif self._metadata['segment_id_type'] == 'UInt16':
+            arr = np.fromstring(arr, dtype=np.uint16)
+        elif self._metadata['segment_id_type'] == 'UInt32':
+            arr = np.fromstring(arr, dtype=np.uint32)
+        elif self._metadata['segment_id_type'] == 'UInt64':
+            arr = np.fromstring(arr, dtype=np.uint64)
+        arr = arr.reshape(self._metadata['chunk_voxel_dimensions'][::-1])
+        return arr.transpose((2,1,0))
```

Comment:
I think you can say `arr.T` if you're not trying to do some kind of special rearrangement.